File: crowdin/src/main/java/com/crowdin/platform/repository/remote/RemoteStringRepository.java
package com.crowdin.platform.repository.remote;

import android.content.Context;

import com.crowdin.platform.api.CrowdinApi;
import com.crowdin.platform.repository.LanguageDataCallback;

public class RemoteStringRepository implements RemoteRepository {

    private final CrowdinApi crowdinApi;

    public RemoteStringRepository(CrowdinApi crowdinApi) {
        this.crowdinApi = crowdinApi;
    }

    @Override
    public void fetchData(Context context, String language, LanguageDataCallback languageDataCallback) {
        // TODO: API call
//        new StringsLoaderTask(context, languageDataCallback).run();
    }
}

File: crowdin/src/main/java/com/crowdin/platform/repository/remote/RemoteStringRepository.kt
package com.crowdin.platform.repository.remote

import android.content.Context

import com.crowdin.platform.api.CrowdinApi
import com.crowdin.platform.repository.LanguageDataCallback

internal class RemoteStringRepository(private val crowdinApi: CrowdinApi) : RemoteRepository {

    override fun fetchData(context: Context, language: String, languageDataCallback: LanguageDataCallback) {
        // TODO: API call
        StringsLoaderTask(context, languageDataCallback).run()
    }
}
